rpcs3/Emu/RSX/VK/vkutils/device.cpp

```diff
@@ -452,6 +452,18 @@ namespace vk
 			enabled_features.shaderStorageImageWriteWithoutFormat = VK_FALSE;
 		}
 
+		if (!pgpu->features.shaderClipDistance)
+		{
+			rsx_log.error("Your GPU does not support shader clip distance. Graphics will not render correctly.");
+			enabled_features.shaderClipDistance = VK_FALSE;
+		}
+
+		if (!pgpu->features.shaderStorageBufferArrayDynamicIndexing)
```

Comment:
lavapipe has this implemented on mesa-git and will be part of Mesa 22.0 release in February. It's best if you test lavapipe with mesa-git, Mesa's current stable has an issue that crashes most games on lavapipe (https://gitlab.freedesktop.org/mesa/mesa/-/issues/4855)

Reply:
I tested on ubuntu 21.10 aarch64 and it does have all this stuff working. For some reason neko's system does not work right, but yea, this is not really a fix. We do invariably use these features unconditionally and there is no reasonable workaround for it as it is core fixed function stuff. I'll re-review when this is all ready for review.

Reply:
I use Ubuntu 22 container which seems to have outdated Mesa but there is no PPA for this version yet or I couldn't find it.